test/e2e/k8s_provisioner_test.go

```diff
@@ -70,5 +79,47 @@ var _ = Describe("k8s provisioner", func() {
 				}).Should(BeTrue())
 			})
 		})
+
+		It("should re-create underlying system resources", func() {
+			var (
+				podNN types.NamespacedName
+			)
+
+			By("getting the underlying bundle unpacking pod")
+			Eventually(func() bool {
+				requirement, err := labels.NewRequirement("core.rukpak.io/owner-name", selection.Equals, []string{bundle.GetName()})
+				if err != nil {
+					return false
+				}
+				pods := &corev1.PodList{}
+				if err := c.List(context.Background(), pods, &client.ListOptions{
+					LabelSelector: labels.NewSelector().Add(*requirement),
+					Namespace:     "rukpak-system",
+				}); err != nil {
+					return false
+				}
+				if len(pods.Items) != 1 {
+					return false
+				}
+				podNN = client.ObjectKeyFromObject(&pods.Items[0])
+				return true
```

Comment:
Yeah honestly this was my first iteration. I'm not entirely sure why I settled on using a podNN vs. a pod variable but it was also 3am in the morning, so it's possible my brain wasn't working at 100% capacity. It seems like this would allow us to eliminate a single call, so it seems worthwhile to change.

Reply:
Updated - note: I had to wrap the delete call in an eventually handler as to avoid 1/3 runs being flaky. This seems like a reasonable workaround to me?